march_joint_inertia_controller/include/march_joint_inertia_controller/joint_trajectory_inertia_controller.h

```diff
@@ -0,0 +1,211 @@
+// Copyright 2020 Project March.
+
+#ifndef MARCH_HARDWARE_JOINT_TRAJECTORY_INERTIA_CONTROLLER_H
+#define MARCH_HARDWARE_JOINT_TRAJECTORY_INERTIA_CONTROLLER_H
+
+#include <cassert>
+#include <control_toolbox/pid.h>
+#include <controller_interface/controller.h>
+#include <hardware_interface/joint_command_interface.h>
+#include <hardware_interface/joint_state_interface.h>
+#include <joint_trajectory_controller/joint_trajectory_controller.h>
+#include "march_joint_inertia_controller/inertia_estimator.h"
+#include <memory>
+#include <pluginlib/class_list_macros.h>
+#include <realtime_tools/realtime_buffer.h>
+#include <realtime_tools/realtime_publisher.h>
+#include <ros/node_handle.h>
+#include <ros/time.h>
+#include <trajectory_interface/quintic_spline_segment.h>
+
+namespace joint_trajectory_controller
+{
+typedef trajectory_interface::QuinticSplineSegment<double> SegmentImpl;
+typedef JointTrajectorySegment<SegmentImpl> Segment;
+typedef typename Segment::State State;
+}  // namespace joint_trajectory_controller
+
+template <>
+class HardwareInterfaceAdapter<hardware_interface::EffortJointInterface, joint_trajectory_controller::State>
+{
+public:
+  HardwareInterfaceAdapter() : joint_handles_ptr_(nullptr)
+  {
+  }
+
+  /**
+   * \brief Initialize the controller by establishing the pointer to the joints, the pid calculators and the inertia
+   * estimators
+   */
+  bool init(std::vector<hardware_interface::JointHandle>& joint_handles, ros::NodeHandle& nh)
+  {
+    joint_handles_ptr_ = &joint_handles;
+    const unsigned int num_joints_ = joint_handles_ptr_->size();
```

Comment:
You shadow the private member `num_joints_` here. So, you either meant to use that one or you should rename it